set default compression to zstd

rayz opened the pull request:
<!--
* New contributors are highly encouraged to read our
  [CONTRIBUTING](/CONTRIBUTING.md) documentation.
* The pull request:
  * Should only fix one issue or add one feature at a time.
  * Must update the test suite for the relevant functionality.
  * Should pass all status checks before being reviewed or merged.
* Commit titles should be prefixed with general area of pull request's change.

-->
### What does this PR do?
Add ability to set zstd as the compression mechanism for payloads.  This is set in `config.go` with `config.BindEnvAndSetDefault("serializer_compressor_kind", {zlib | zstd})`
<!--
* A brief description of the change being made with this pull request.
* If the description here cannot be expressed in a succinct form, consider
  opening multiple pull requests instead of a single one.
-->

### Motivation

<!--
* What inspired you to submit this pull request?
* Link any related GitHub issues or PRs here.
-->

### Additional Notes
For first release keep `zlib` the default compression. Internally we will deploy it with the config set to zstd. If all looks good then the next Agent release we can change the default to be `zstd`

<!--
* Anything else we should know when reviewing?
* Include benchmarking information here whenever possible.
* Include info about alternatives that were considered and why the proposed
  version was chosen.
-->

### Possible Drawbacks / Trade-offs

<!--
* What are the possible side-effects or negative impacts of the code change?
-->

### Describe how to test/QA your changes

<!--
* Write here in detail or link to detailed instructions on how this change can
  be tested/QAd/validated, including any environment setup.
-->

### Reviewer's Checklist
<!--
* Authors can use this list as a reference to ensure that there are no problems
  during the review but the signing off is to be done by the reviewer(s).

Note: Adding GitHub labels is only possible for contributors with write access.
-->

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

hush-hush committed:
Extract helper function from the config pkg

Moving pure helper function to pkg/config/utils allow to use them with comp/core/config comp

0intro committed:
[omnibus] Upgrade OpenSCAP to version 1.3.9 (#19302)

This change upgrades OpenSCAP to version 1.3.9.
Consequently, we remove the backported fixes,
along with the changes that were merged upstream.

Moreover, the dependency on the PCRE library has been
replaced by the PCRE2 library.

katiehockman committed:
trace: Enable DB statement obfuscation in span tags by default (#19321)

Enable DB statement obfuscation for span tags by default which was previously off by default.
This affects MongoDB queries, ElasticSearch request bodies, and raw commands from Redis and MemCached

paulcacheux committed:
[CWS] ensure network context is scoped to DNS events (#19317)

* ensure network context is only available on DNS events

* generate documentation

rayz committed:
draft zstd

rayz committed:
update test

rayz committed:
update new compressor params

rayz committed:
add nil checks

FlorentClarret committed:
Add aerospike to ARM platforms (#19205)

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go-v2/service/ec2 from 1.63.1 to 1.117.0 (#19319)

* Bump github.com/aws/aws-sdk-go-v2/service/ec2 from 1.63.1 to 1.117.0

Bumps [github.com/aws/aws-sdk-go-v2/service/ec2](https://github.com/aws/aws-sdk-go-v2) from 1.63.1 to 1.117.0.
- [Release notes](https://github.com/aws/aws-sdk-go-v2/releases)
- [Commits](https://github.com/aws/aws-sdk-go-v2/compare/service/ec2/v1.63.1...service/ec2/v1.117.0)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go-v2/service/ec2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Auto-generate go.sum and LICENSE-3rdparty.csv changes

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go-v2/credentials from 1.13.26 to 1.13.37 (#19335)

Bumps [github.com/aws/aws-sdk-go-v2/credentials](https://github.com/aws/aws-sdk-go-v2) from 1.13.26 to 1.13.37.
- [Release notes](https://github.com/aws/aws-sdk-go-v2/releases)
- [Commits](https://github.com/aws/aws-sdk-go-v2/compare/credentials/v1.13.26...credentials/v1.13.37)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go-v2/credentials
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/hashicorp/golang-lru/v2 from 2.0.5 to 2.0.6 (#19020)

Bumps [github.com/hashicorp/golang-lru/v2](https://github.com/hashicorp/golang-lru) from 2.0.5 to 2.0.6.
- [Release notes](https://github.com/hashicorp/golang-lru/releases)
- [Commits](https://github.com/hashicorp/golang-lru/compare/v2.0.5...v2.0.6)

---
updated-dependencies:
- dependency-name: github.com/hashicorp/golang-lru/v2
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go from 1.44.320 to 1.45.5 (#19338)

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.44.320 to 1.45.5.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.44.320...v1.45.5)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/invopop/jsonschema from 0.7.0 to 0.8.0 (#19301)

Bumps [github.com/invopop/jsonschema](https://github.com/invopop/jsonschema) from 0.7.0 to 0.8.0.
- [Commits](https://github.com/invopop/jsonschema/compare/v0.7.0...v0.8.0)

---
updated-dependencies:
- dependency-name: github.com/invopop/jsonschema
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

pgimalac committed:
fix(test/benchmarks): fix some import path (#19284)

dependabot[bot] committed:
Bump github.com/hashicorp/golang-lru/v2 from 2.0.5 to 2.0.6 in /pkg/security/secl (#19064)

Dependabot couldn't find the original pull request head commit, c5c546a87368de66d2ec44c4d46c24c77f59d5e3.

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

spikat committed:
[CWS] Fix crash when trying to save unavailable profile (#19024)

paulcacheux committed:
allow failure of CWS e2e tests (#19341)

AliDatadog committed:
[workloadmeta] Make kubeapiserver collector more generic (#18991)

nenadnoveljic committed:
Fix SQL text truncation in Oracle query samples (#19316)

* fixed

* rel notes

* relnotes

* constants

* corrections

* error fixed

* linter errors

* linter errors

nenadnoveljic committed:
Improve Oracle example configuration file (#19264)

* example conf updates

* rel note

* Update cmd/agent/dist/conf.d/oracle-dbm.d/conf.yaml.example

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

* Update cmd/agent/dist/conf.d/oracle-dbm.d/conf.yaml.example

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

---------

Co-authored-by: Ursula Chen <58821586+urseberry@users.noreply.github.com>

lizaerem committed:
Add --use-unconnected-udp-socket flag to agent snmp walk command (#19333)

* Add --use-unconnected-udp-socket flag to agent snmp walk command

* Add release note

* Update release note (name too long)

* Update release note

just-chillin committed:
[PROCS-2802][process-agent] Fix flaky `TestStatsWithPermByPID` (#19256)

* Fix flaky test

* Add require

* Fix tests

* Fix remaining tests

* Fix linter errors

* Fix tests

* `WithCustomeProcFSRoot` => `WithProcFSRoot`

hmahmood committed:
[NPM] Add log to debug test failure (#19349)

rayz committed:
refactor tests to include zstd

JLineaweaver committed:
[orchestrator] Add tags to cr/crds (#19348)

ajgajg1134 committed:
pkg/trace/agent: Replace Tags now acts on 'metrics' tags (#19350)

* pkg/trace/agent: Replace Tags now acts on 'metrics' tags
* add release notes
* refactor duplicate code
* Add some unchanging metrics and meta to ensure no inadvertant changes

DylanLovesCoffee committed:
[serverless] Fix flaky logs_collector unit test (#19344)

fix flaky unit test

blt committed:
Expand the UDS/DogStatsD experiment set (#19324)

* Expand the UDS/DogStatsD experiment set

This commit updates the `uds_dogstatsd_to_api` to conform to the behavior in the
`_nodist` variant excepting that we maintain the ratio of metrics in the
`app_server` workload check. We also introduce a `_nomultivalue` variant of the
baseline experiment that is intended to probe Agent's behavior when multiple
values are not packed into the same metric payload.

REF SMP-664

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* re-introduce service and event checks

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

* correct profiling tag name

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

---------

Signed-off-by: Brian L. Troutwine <brian.troutwine@datadoghq.com>

guyarb committed:
usm: process monitor: Remove error log (#19361)

chouetz committed:
Adapt the agent-platform jira project after migration (#19315)

* Adapt the agent-platform jira project after migration

* Update ddqa config with new project names as well

paulcacheux committed:
[CWS] add `PostProbeStart` callback for event consumers, and run self tests in it (#19328)

* add `AfterProbeStart` for event consumers, and run self tests in it

* use an ad-hoc interface instead

L3n41c committed:
Fix `hacky_dev_image_build` invoke task (#19323)

CONTINT-636

paulcacheux committed:
Revert "allow failure of CWS e2e tests (#19341)" (#19368)

This reverts commit 9b4940b7246afa03b4cdbfc132b4a698d009801d.

guyarb committed:
usm: fix ifndef (#19367)

* usm: tls: Fixed ifndef

* usm: unify ifndef format

guyarb committed:
usm: disable all irrelevant maps (#19366)

vickenty commented on pkg/serializer/internal/stream/zipper.go:
```diff
@@ -0,0 +1,39 @@
+}
+
+// NewZipperWriter returns a zipper of either zstd or zlib depending on the kind param
+func NewZipperWriter(kind string, output *bytes.Buffer) (Zipper, error) {
```
Would it make sense to match the function name with the returned type name?

```suggestion
func NewZipper(kind string, output *bytes.Buffer) (Zipper, error) {
```

Also, does it (and the interfaces) need to be public?

amenasria committed:
[CI] Remove go linters for `rpm`, `amd64` and `arm64`. (#18923)

rayz committed:
move zipper to compressor and update tests

rayz commented on pkg/serializer/internal/stream/compressor_test.go:
```diff
@@ -90,110 +109,203 @@ func TestCompressorAddItemErrCodeWithEmptyCompressor(t *testing.T) {
+		maxPayloadSize int
+	}{
+		"zlib": {kind: "zlib", maxPayloadSize: 22},
+		"zstd": {kind: "zstd", maxPayloadSize: 20}, // zstd more efficient
```
yea, ` zstd` was returning `{[A,B,C,D]}` in payloads[0] when it had the same `maxPayloadSize` as `zlib`

KevinFairise2 committed:
AETT-114 Create function to copy file and folder through ssh on remote host (#19223)

* Create function to copy file and folder through ssh on remote host

* Fix comments

* Go mod tidy

* Address suggestions

chouetz committed:
Use cache by default for python requirements (#17180)

hmahmood committed:
Bump memory for kitchen test jobs (#19389)

guyarb committed:
usm: http: minimize http_transaction struct by reorder the inner fields (#19363)

github-actions[bot] committed:
CWS: sync BTFhub constants (#19401)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

chouetz committed:
Trace service start/stop logs in case of error (#19097)

* Trace service start/stop logs in case of error

* Factorize code

nenadnoveljic committed:
fixed (#19373)

dependabot[bot] committed:
Bump golang.org/x/tools from 0.12.0 to 0.13.0 in /pkg/security/secl (#19399)

* Bump golang.org/x/tools from 0.12.0 to 0.13.0 in /pkg/security/secl

Bumps [golang.org/x/tools](https://github.com/golang/tools) from 0.12.0 to 0.13.0.
- [Release notes](https://github.com/golang/tools/releases)
- [Commits](https://github.com/golang/tools/compare/v0.12.0...v0.13.0)

---
updated-dependencies:
- dependency-name: golang.org/x/tools
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

* Auto-generate go.sum and LICENSE-3rdparty.csv changes

---------

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>
Co-authored-by: dependabot[bot] <dependabot[bot]@users.noreply.github.com>

pducolin committed:
[e2e] run remove stack command from temp dir (#19382)

* [e2e] run remove stack command from temp dir

* [e2e] use json format to print pulumi about

pducolin committed:
[e2e] move examples to testify eventually (#19391)

KevinFairise2 committed:
Add example for ex 8 (#19392)

* Add example for ex 8

* Change ex8 to ex7

0intro committed:
[omnibus] Reduce memory footprint of dpkginfo probe in OpenSCAP (#19409)

This change adds a patch to OpenSCAP to remove the call
to the cgCache->ReadOnlyOpen function in dpkginfo probe.

Calling this function is not useful in our case and
causes memory allocations that could be avoided.

vickenty committed:
Add distribution support to corechecks (#19074)

KevinFairise2 committed:
Edit copy example to match ex 6 (#19380)

* Edit copy example to match ex 6

* Add suggestions from previous PR

* Last fix to example 6

KevinFairise2 committed:
AETT-128 Add tasks to wrap aws-vault on new-e2e go tests (#19080)

* Add tasks to wrap aws-vault on new-e2e go tests

* Remove env file since it looks workin without it

* Add gitignore

paulcacheux committed:
[CWS] LRU based string interning for args and envs values (#19385)

* best effort string interning of args and envs values

* fix locking issue

* add comments

* bump the cache size and use constant

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go from 1.45.5 to 1.45.7 (#19396)

Bumps [github.com/aws/aws-sdk-go](https://github.com/aws/aws-sdk-go) from 1.45.5 to 1.45.7.
- [Release notes](https://github.com/aws/aws-sdk-go/releases)
- [Commits](https://github.com/aws/aws-sdk-go/compare/v1.45.5...v1.45.7)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nenadnoveljic committed:
[Oracle] Detect missing privileges (#19381)

* oracle-missing-privileges

* release note

* linter

* linter errors

* Update releasenotes/notes/oracle-privilege-info-f66e19ac7c8aff57.yaml

Co-authored-by: Heston Hoffman <hestonhoffman@gmail.com>

---------

Co-authored-by: Heston Hoffman <hestonhoffman@gmail.com>

Hellzy committed:
pkg/serverless: log context.Canceled errors as Debug instead of Error (#19408)

pgimalac committed:
Bump go to 1.20.8 (#19339)

pgimalac committed:
[ASCII-737] Add an invoke task to automate go updates (#19347)

* feat: add invoke task to automate go updates

* add a warn flag and exit by default in case of mismatch, use semver to parse version

* move main function at the top

* fix import issues

* fix review comment

nenadnoveljic committed:
custom queries for non-DBM users (#19402)

p-lambert committed:
Revert "usm: fix incomplete buffer timestamp parameter (#18928)" (#19425)

This reverts commit 602a33f2fc1c9a99689e4156ce76877752e45485.

ogaca-dd committed:
[Component framework]: Create a minimal demultiplexer component (#19214)

* Add demultiplexer component

* inject sender.SenderManager in diagnoseChecksInCLIProcess

* Remove aggregator.GetSender

* Use NoopSenderManager in unit tests when there is no aggregator

* Use SenderManager from NewMockSender when possible

* Remove remaining aggregator.GetSenderManager() from tests

* Fix tests conditioned by a build flag + fix comp/aggregator.

* rename alias

* Fix integration tests

gh123man commented on pkg/serializer/internal/stream/compressor_test.go:
```diff
@@ -90,110 +109,203 @@ func TestCompressorAddItemErrCodeWithEmptyCompressor(t *testing.T) {
+			require.Len(t, payloads, 1)
+
+			// r, err := tc.r()
+			// require.NoError(t, err)
```
(also appears in several other files - so we don't lose track)

mftoure committed:
Add image_id tag to container entity (#19410)

* Add image_id tag to container entity

* Add changelog

hush-hush committed:
Adding a mock to the resources comp

AliDatadog committed:
generate proto message for core-agent <> cluster-agent communication (#19383)

paulcacheux committed:
CI: bump buildimages to include the goroot fix (#19433)

paulcacheux committed:
[CWS] add config toggle to use own CWS track (#19192)

chouetz committed:
Simplify python requirements (#19288)

* Use the uploader image to remove duplication in python requirements

* [gitlab] migrate notify stage to k8s runners

* Use new images with correct requirements.txt

* update to last version

---------

Co-authored-by: pducolin <paola.ducolin@datadoghq.com>

guyarb committed:
usm: tls: rework read buffer (#19421)

0intro committed:
[omnibus] Update dependencies of OpenSCAP (#19437)

This change updates dependencies of OpenSCAP.
It includes the following software definitions:

 - apt
 - dbus
 - elfutils
 - file
 - glib
 - gmp
 - gnutls
 - libgcrypt
 - libgpg-error
 - libselinux
 - libsepol
 - libxxhash
 - lua
 - nettle
 - rpm
 - sqlite
 - util-linux
 - xmlsec
 - zstd

katiehockman committed:
pkg/trace: ensure the sql.query tag always contains the obfuscated value (#19354)

The trace-agent will always set the sql.query tag to an obfuscated value to avoid exposing sensitive data. This is a change from the previous behavior where the trace-agent would not change the sql.query tag if it was already set.

modernplumbing committed:
[CWS] CWS RC true unless explicitly false or globally disabled (#19357)

* cws rc true unless explicitly false or globally disabled

* release note

* linter change

* disable rc in module tester

hush-hush committed:
Split flarehelpers interface from implementation (#19420)

A component that supports adding information to a flare will not longer
pull all the flare dependencies when included in a binary that doesn't
support flare creation.

YoannGh committed:
[CWS] run field checks when calling GetFieldValue (#19419)

* [CWS] run field checks when calling GetFieldValue

* return error instead of nil when the check doesn't pass

* run checks in SetFieldValue as well

* remove tmpfile

* fix model unit tests and TestProcessInterpreter functional tests

brycekahle committed:
Add support for custom contexts to profiling, include system-probe module name (#19330)

* Add ability to set profiler custom context

* Add module pprof label for system-probe modules

* Add gRPC support

* add labels to GetStats and Close

* fix error condition

safchain committed:
[CWS] make use of device for the mount resolution (#18918)

* [CWS] make use of device for the mount resolution

* use redemption in finalize for children

safchain committed:
[CWS] on-demand SECL Partial (#19386)

safchain committed:
[CWS] use 64bits cookie (#19149)

just-chillin committed:
Enable TLS (#19451)

brycekahle committed:
Ignore missing PR error from pr-commenter (#19353)

* ignore no PR error from pr-commenter

* do not error immediately from command

brycekahle committed:
move CWS system-probe config into separate file (#19428)

hmahmood committed:
Revert "[NPM] Add log to debug test failure (#19349)" (#19393)

This reverts commit ac0b07839c9a7aae13a60147a3f53f13eb954b10.

hmahmood committed:
Only count evicts on adding to process cache (#19388)

pgimalac committed:
[ASCII-717] Avoid using gotest.tools/assert (#19422)

* refactor: do not use gotest.tools/assert

* add better comment in .golangci.yml

github-actions[bot] committed:
CWS: sync BTFhub constants (#19456)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

dependabot[bot] committed:
Bump github.com/aws/aws-sdk-go-v2/service/ec2 from 1.117.0 to 1.118.0 (#19454)

Bumps [github.com/aws/aws-sdk-go-v2/service/ec2](https://github.com/aws/aws-sdk-go-v2) from 1.117.0 to 1.118.0.
- [Release notes](https://github.com/aws/aws-sdk-go-v2/releases)
- [Commits](https://github.com/aws/aws-sdk-go-v2/compare/service/ec2/v1.117.0...service/ec2/v1.118.0)

---
updated-dependencies:
- dependency-name: github.com/aws/aws-sdk-go-v2/service/ec2
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

chouetz committed:
Change windows instance (#19436)

* Change windows instance

* Use another region instead

nenadnoveljic committed:
[oracle] Emit zero value for pga_over_allocation (#19438)

* emit 0 value for pga_over_allocation

* relnote

* linter

nenadnoveljic committed:
[Oracle] Allow unobfuscated plan logging (#19435)

* Add a parameter for logging unobfuscated plan information

* testing error

* copy/paster error fix

* use error instead of errorf

nenadnoveljic committed:
[Oracle] Add network disruption handling (#19432)

* add refused connection handling

* relnote

* linter

* Update releasenotes/notes/oracle-handling-refused-connection-8606661daa27952c.yaml

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

---------

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

rayz committed:
attempt refactor compression headers

jennchenn committed:
[DCA] Increment `admission_webhooks.mutation_attempts` metric on lib injection (#19387)

* Update mutation attempts metric to support lib injection

* Increment mutation attempt metric for lib injections

* Add release note to document changes

* Shorten release note filename

* Log error when injectLibRequirements fails

jinroh committed:
[CSPM] Add capability to scan Linux packages (#19077)

* [CSPM] Add capability to scan Linux packages

Adds a new `package` fields to rules inputs specs. This package field can list
software / package names that our resolver will try to match with installed
software on the host.

We only support apt (dpkg) / rpm and apk package managers.

If a package match one of the names, it can be used directly from our rego
rules and should contain version and architecture metadata.

* fix reading dpkg status.d

paulcacheux committed:
bump security agent policies to v0.48.1 (#19459)

chouetz committed:
Add a method to retrieve a jira card for failed test (#19305)

rayz committed:
attempt refactor compression headers

Commit:
Changed http2 map size to 1024 (#19462)

jinroh committed:
[CSPM] Rely on PID instead of cgroup/container ID for resolver (#19465)

This helps avoiding having to re-scan the proc table, since the PID is the
only piece of data required here to calculate the host root path and retrieve
the associate cgroup / container ID.

nenadnoveljic committed:
[Oracle] Adding the hosting-type tag (#19440)

* Add hosting-type tag

* relnotes

* Update releasenotes/notes/oracle-hosting-type-tag-c2ad1ebfc55fb7c9.yaml

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

* changed hosting-type to hosting_type

* linter

* linter

* corrected pga_over_allocation_count after teh merge

---------

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

brycekahle committed:
Use table formatting for PR comments (#19122)

safchain committed:
Add uds and extra tags for the internal profiler (#19453)

Gui774ume committed:
[CWS] Enable hash resolver by default (#19415)

hokitam committed:
Add initialization debug logs to DSM pipeline_stats handler in Datadog agent (#19429)

* Adding more debug logs to pipeline_stats handler in agent

* Add more debug log per call to DSM intake

* update error msg

Yumasi committed:
USM: add uprobe tail calls facilities (#19461)

* USM: add uprobe tail calls facilities

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

* fix variable's names

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

* add docs

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

---------

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>
Co-authored-by: Guy Arbitman <guy20495@gmail.com>

hmahmood committed:
Fix enforcing DNS stats limit (#19262)

akarpz committed:
Force install of nf_conntrack_netlink before creation of ebpf_conntracker (#18970)

* retry creation of ebpf_conntracker

* dump table before creating conntracker

* manually add probe to kernel before attempting to create ebpf_conntracker

* remove debug

* dont return early if modprobe fails

* only try loading ebpf conntracker if modprobe call succeeds

* don't log modprobe failure explicitly

gh123man committed:
[Logs Agent Component] Migrate top level logs agent implementation to `comp` (#18743)

* Migrate logs agent internals to `comp`

* Update comp/logs/agent/agent.go

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

* linter errors

* Log error in serverless when logs agent does not start

---------

Co-authored-by: Vickenty Fesunov <vickenty@users.noreply.github.com>

github-actions[bot] committed:
CWS: sync BTFhub constants (#19484)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

nenadnoveljic committed:
[oracle] Improve SQL text collection for activity samples (#19431)

* wip

* improve SQL text collection

* relnote

* more info in release notes

* warning to debug

* don't obfuscate bg process name

* Update releasenotes/notes/oracle-enrich-sqltext-aa778da34a8eb4ed.yaml

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

* relnote correcion

---------

Co-authored-by: May Lee <mayl@alumni.cmu.edu>

AlexandreYang committed:
[pkg/networkdevice] Modularise SNMP profiles data model (#19049)

dependabot[bot] committed:
Bump datadog-api-client from 2.16.0 to 2.17.0 in /test/e2e/cws-tests (#19486)

Bumps [datadog-api-client](https://github.com/DataDog/datadog-api-client-python) from 2.16.0 to 2.17.0.
- [Release notes](https://github.com/DataDog/datadog-api-client-python/releases)
- [Changelog](https://github.com/DataDog/datadog-api-client-python/blob/master/CHANGELOG.md)
- [Commits](https://github.com/DataDog/datadog-api-client-python/compare/2.16.0...2.17.0)

---
updated-dependencies:
- dependency-name: datadog-api-client
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

nplanel committed:
TestE2E GetCollectedTimeByName (#19390)

Return the timestamp when the payloads has been collected aim to check if it was received as expected.

AlexandreYang committed:
[pkg/networkdevices] Add omitempty to Profile Definition (#19340)

dependabot[bot] committed:
Bump pyaml from 23.7.0 to 23.9.6 in /test/e2e/cws-tests (#19487)

Bumps [pyaml](https://github.com/mk-fg/pretty-yaml) from 23.7.0 to 23.9.6.
- [Commits](https://github.com/mk-fg/pretty-yaml/commits)

---
updated-dependencies:
- dependency-name: pyaml
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

davidor committed:
[corechecks/docker] Convert CPU weight to shares in docker.cpu.shares (#19474)

* [util/cgroups] Separate CPU shares and weight

* [util/containers/metrics] Separate CPU shares and weight

* [corechecks/docker] Convert CPU weight to shares in docker.cpu.shares

* Add release note

* [corechecks/containers] Don't emit docker.cpu.shares if it's 0

davidor committed:
[utils/ecs/metadata/detection] Use client URL directly (#19496)

robertjli committed:
[process-agent] Add workloadmeta extractor stats to status output (#19482)

nplanel committed:
fix Test e2e if OS is not linux (#19500)

Seems some OS doesn't have monotonic implementation of time.Now()

davidor committed:
[utils/ecs/metadata/testutil/dummy_ecs] Simplify Start() (#19501)

The port returned is no longer used anywhere.

safchain committed:
[CWS] use a single empty process cache entry (#19493)

hmahmood committed:
[NPM] Fix dns stats counters (#19483)

Commit:
pkg/trace/api: Add debugger diagnostics proxy endpoint (#18501)

* Add debugger diagnostics proxy endpoint in trace-agent

* Add release note

* Add doc comments; rename variable

* Update release note with more information

* Rename intakeDDTagsQueryStringMaximumLength; add doc comment

* Rename variable to ddTagsQueryStringMaxLen; update log message

* Add clarifying comment for ddTagsQueryStringMaxLen

modernplumbing committed:
[CWS] Benchmark test for dispatching events to Event Monitor consumers (#19356)

* benchmark test

* add b.Run block

* put wildcard events in separate func to make profiling easier

guyarb committed:
usm: tls: introduce tls process (#19362)

Yumasi committed:
USM: grpc: increment data offset properly when skipping a header (#19210)

* grpc: increment data offset properly when skipping a header

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

* tests: add docs

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

---------

Signed-off-by: Guillaume Pagnoux <guillaume.pagnoux@datadoghq.com>

github-actions[bot] committed:
CWS: sync BTFhub constants (#19508)

Co-authored-by: paulcacheux <paulcacheux@users.noreply.github.com>

nenadnoveljic committed:
[Oracle] Improve logging (#19475)

* logging improved

* relnote

* construct prompt from login

* rebased to main to resolve conflicts

* linter

* rebased to main bc of conflicts

* fix tests

* linter

* refine no sql text found message

* parsing lost connection for go ora

nenadnoveljic committed:
failing tests removed (#19495)

safchain committed:
[CWS] disable network when CWS is not enabled (#19498)

gbbr committed:
comp/otelcol: add collector component and otel-agent binary (#18360)

This change migrates the pkg/otlp collector to a reusable component and adds a test binary to validate reusability. It is still an intermediate step because some related items are not yet migrated to components (e.g. pkg/status) which continue to require the global state, and others are migrated to components but can not be depended on (e.g. logs component) because of lazy initialisations required by the collector (the logs channel).

ahmed-mez committed:
pkg/trace/api: fix flaky TestHTTPReceiverStart (#19513)

KevinFairise2 committed:
Release 1.3 version of the MacOs install script (#19502)

rayz committed:
attempt refactor compression headers 3
